.github/workflows/deploy.yml

```diff
@@ -1,13 +1,37 @@
-# This is a copy-and-past version of ci.yml but additionally creating a release
+# This is a copy-and-paste version of ci.yml but additionally creating a release
 name: Release Artifacts
 
 on:
+  schedule:
+    - cron: '0 3 * * 1'  # Every Monday at 03:00 UTC
   push:
     tags:
       - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
 jobs:
+  calculate-version:
+    name: Calculate Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.set-version.outputs.VERSION }}
+    steps:
+
+      - name: Calculate new version
+        id: set-version
+        run: |
+          if [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_OUTPUT
+          else
+            MAJOR=0
+            YEAR=$(date +%y)
+            WEEK=$(date +%W)
+            WEEK_PADDED=$(printf "%02d" $WEEK)
+            echo "VERSION=${MAJOR}.${YEAR}.${WEEK_PADDED}" >> $GITHUB_OUTPUT
```

Comment:
Do we always want to release as `0.YY.WW.0` so that we can make subsequent releases?